### PR TITLE
Fix MoviesController_Pages_ReturnSuccess test failure

### PR DESCRIPTION
**Pull request by pakrym**
Fixes NRE failure in `MoviesController_Pages_ReturnSuccess`, which was hidden by `UseDeveloperExceptionPage`.

### CODE REVIEW

**Comment by pranavkm** [comment]
It would be nice to fix the pattern we have here and use https://github.com/aspnet/Mvc/blob/dev/test/WebSites/HtmlGenerationWebSite/SignalTokenProviderService.cs to manage the change tokens. (Particularly because this is a sample). Although that might require a bunch of changes.

**Comment by pakrym** [comment]
@pranavkm I'm OK with copying it, but do we want to complicate TagHelperSample with it?

**Comment by pranavkm** [comment]
It's a sample and we should try and make it as close to the pattern we expect users to use.

**Comment by pranavkm** [comment]
cc @NTaylorMullen 

:shipit: run it by @Eilon for ask mode approval.

File: samples/Mvc.TagHelperSample.Web/Services/ISignalTokenProviderService.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Primitives;
+
+namespace TagHelperSample.Web.Services
+{
+    public interface ISignalTokenProviderService<TKey>
```
Comment (NTaylorMullen) [inline samples/Mvc.TagHelperSample.Web/Services/ISignalTokenProviderService.cs]: Doesn't need `TKey`. The implementation can assume the type. It's only ever used for `FeaturedMovies`. No need to make it more complex :smile:

File: samples/Mvc.TagHelperSample.Web/Services/MoviesService.cs [hunk 2]
```diff
@@ -44,15 +55,21 @@ public string GetCriticsQuote(out IChangeToken expirationToken)
                 "Bravo!"
             };
 
-            expirationToken = new CancellationChangeToken(_quotesTokenSource.Token);
-            return quotes[_random.Next(0, quotes.Length)];
+            string quote;
+            if (!_memoryCache.TryGetValue(movieName, out quote))
```
Comment (NTaylorMullen) [inline samples/Mvc.TagHelperSample.Web/Services/MoviesService.cs]: https://github.com/aspnet/Caching/blob/dev/src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs#L98

File: samples/Mvc.TagHelperSample.Web/Services/MoviesService.cs [hunk 3]
```diff
@@ -5,37 +5,48 @@
...
+            _memoryCache = memoryCache;
+            _tokenProviderService = tokenProviderService;
+        }
 
-            expirationToken = new CancellationChangeToken(_featuredMoviesTokenSource.Token);
-            return GetMovies().OrderBy(m => m.Rank).Take(2);
+        public IEnumerable<FeaturedMovies> GetFeaturedMovies()
+        {
+            IEnumerable<FeaturedMovies> movies;
+            if (!_memoryCache.TryGetValue(FeaturedTokenKey, out movies))
```
Comment (NTaylorMullen) [inline samples/Mvc.TagHelperSample.Web/Services/MoviesService.cs]: https://github.com/aspnet/Caching/blob/dev/src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs#L98

File: samples/Mvc.TagHelperSample.Web/Services/SignalTokenProviderService.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+using Microsoft.Extensions.Primitives;
+
+namespace TagHelperSample.Web.Services
+{
+    public class SignalTokenProviderService : ISignalTokenProviderService
+    {
+        private readonly ConcurrentDictionary<object, ChangeTokenInfo> _changeTokens
+            = new ConcurrentDictionary<object, ChangeTokenInfo>();
+
+        public IChangeToken GetToken(object key)
```
Comment (NTaylorMullen) [inline samples/Mvc.TagHelperSample.Web/Services/SignalTokenProviderService.cs]: We only ever pass in string keys, might be worth simplifying this to express that.

File: samples/Mvc.TagHelperSample.Web/Services/MoviesService.cs (hunk 3 above)
Comment (pakrym) [inline samples/Mvc.TagHelperSample.Web/Services/MoviesService.cs]:
```
 public IEnumerable<FeaturedMovies> GetFeaturedMovies()
        {
            var movies = _memoryCache.GetOrCreate(FeaturedTokenKey,
                (entry) =>
                {
                    var value = GetMovies().OrderBy(m => m.Rank).Take(2);

                    entry.SetValue(value).SetOptions(new MemoryCacheEntryOptions()
                        .AddExpirationToken(_tokenProviderService.GetToken(FeaturedTokenKey)));
                    return value;
                }
            );
            return movies;
        }
```

Looks bad

Comment (pakrym) [inline samples/Mvc.TagHelperSample.Web/Services/MoviesService.cs]: Nevermind, did it better

**Comment by pakrym** [comment]
@NTaylorMullen done

**Comment by NTaylorMullen** [comment]
:shipit: after approval

**Pull request merged**